File: Back/Models/problem.js
const mongoose = require('mongoose');

const problemContentSchema = new mongoose.Schema({
  problemId: Number,
  content: String,
  numberOfBlanks: Number,
});

const problemChoiceSchema = new mongoose.Schema({
  choiceId: Number,
  problemId: Number,
  choiceText: String,
});

const problemAnswerSchema = new mongoose.Schema({
  answerId: Number,
  problemId: Number,
  blankPosition: Number,
  correctChoice: problemChoiceSchema,
  correctAnswerText: String,
});

const problemSchema = new mongoose.Schema({
  problemId: Number,
  title: String,
  problemType: String,
  category: String,
  difficulty: Number,
  problemContent: problemContentSchema,
  problemChoices: [problemChoiceSchema],
  problemAnswers: [problemAnswerSchema],
});

module.exports = mongoose.model('Problem', problemSchema);

File: backend/battle/src/main/java/com/scf/battle/domain/dto/Problem/Problem.java
package com.scf.battle.domain.dto.Problem;

import java.util.List;

import com.scf.battle.domain.enums.ProblemType;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

@Data
@NoArgsConstructor
public class Problem {

    private Long problemId;
    private String title;
    private ProblemType problemType;
    private String category;
    private int difficulty;
    private ProblemContent problemContent;
    private List<ProblemChoice> problemChoices;
    private List<ProblemAnswer> problemAnswers;

    @Builder
    public Problem(Long problemId, String title, ProblemType problemType, String category,
        int difficulty, ProblemContent problemContent, List<ProblemChoice> problemChoices,
        List<ProblemAnswer> problemAnswers) {
        this.problemId = problemId;
        this.title = title;
        this.problemType = problemType;
        this.category = category;
        this.difficulty = difficulty;
        this.problemContent = problemContent;
        this.problemChoices = problemChoices;
        this.problemAnswers = problemAnswers;
    }
}
